### PR TITLE
[PROGRAMMES-6534][fix] -  V3 solution (There is another fix for V2)

### PR DESCRIPTION
**Pull request by tatitati**
Because this issue is the same that is being fixed also for V2 right now I suggest you to have a look also here:
https://github.com/bbc/programmes/pull/884

As you can see we consider that every time that a profile has a parent, this come with all the information about it, but is not the case. This exceptional situation (when the info is missed) happens when the parent was never published and instead is/was always in progress. As a reference these are the XML responses for two interesting cases.

<details>
  <summary>XML for parent not published ever</summary>

```xml
<!-- 
THIS XML IS AN EXAMPLE OF AN INDIVIDUAL PROFILE PUBLISHED BUT
 WITH THE PARENT PROFILE ALWAYS BEING IN PROGRESS AND NO PUBLISHED 
-->
<?xml version="1.0" encoding="UTF-8"?>
<result xmlns="https://production.bbc.co.uk/isite2/contentreader/xml/result">
	<metadata>
		<modifiedDateTime>2018-08-15T14:29:04.979762Z</modifiedDateTime>
		<creationDateTime>2014-05-20T13:50:45.687017Z</creationDateTime>
		<type>programmes-profile</type>
		<fileId>programmes-profile-b006tp43-samiraahmed</fileId>
		<guid>5531af13-a351-3960-93d1-193ec85bfb95</guid>
	</metadata>
	<document>
		<form xmlns="https://production.bbc.co.uk/isite2/project/progs-radio3/programmes-profile">
			<metadata>
				<name>Samira Ahmed (b006tp43)</name>
				<title>Samira Ahmed updated (bbbbbbbb... in prgress)</title>
				<type>individual</type>
				<short_synopsis>asdfasdfasdfasdf</short_synopsis>
				<long_synopsis>Samira Ahmed is an award winning journalist with 20 years experience in print and broadcast and a Visiting Professor of Journalism at Kingston University.</long_synopsis>
				<image>p00qsl1w</image>
				<position>0</position>
				<tagline/>
				<group_size/>
				<parent_pid>b006tp43</parent_pid>
				<branding_id/>
				<file_id>b006tp43-samiraahmed</file_id>
				<contributor_pid/>
				<contribution_pid/>
				<parents>
                                       <!-- SECTION OF INTEREST -->
					<parent>
						<result xmlns="https://production.bbc.co.uk/isite2/contentreader/xml/result">
							<metadata>
								<fileId>programmes-profile-1534346917</fileId>
							</metadata>
							<document/>  <!-- PARENT WAS NEVER PUBLISHED. IS ONLY IN PROGRESS SO THIS IS EMPTY-->
						</result>
					</parent>
				</parents>
				<bbc_site/>
			</metadata>
			<key_facts/>
			<profile>
				<image_portrait/>
				<content_blocks>
					<content_block>
						<result xmlns="https://production.bbc.co.uk/isite2/contentreader/xml/result">
							<metadata>
								<modifiedDateTime>2014-06-06T13:58:47.22545Z</modifiedDateTime>
								<creationDateTime>2014-05-20T13:50:45.687017Z</creationDateTime>
								<type>programmes-content-prose</type>
								<fileId>programmes-content-b006tp43-samiraahmed-profile-prose-1</fileId>
								<guid>73a2bd99-0be1-3df0-8cf8-050aa660c0d8</guid>
							</metadata>
							<document>
								<form xmlns="https://production.bbc.co.uk/isite2/project/progs-radio3/programmes-content-prose">
									<metadata>
										<name>Samira Ahmed (b006tp43) - Prose 1</name>
										<file_id>b006tp43-samiraahmed-profile-prose-1</file_id>
									</metadata>
									<content>
										<title/>
										<prose>&lt;p&gt;She has fronted The Proms on BBC4, presents Sunday Morning Live and Newswatch on BBC1 and writes a monthly column in The Big Issue. Samira began her career as a BBC news trainee in 1990. She covered the OJ Simpson case as BBC Los Angeles Correspondent, where she made a documentary about the history of the city's cinemas. Her films as a presenter and reporter at Channel 4 News from 2000 to 2011 included assessing social mobility on screen since the 1960s with Julie Walters and Douglas Trumbull's views on ecological disaster from Silent Running to the BP Gulf of Mexico disaster.&lt;/p&gt;&lt;p&gt;She wrote and presented the acclaimed Channel 4 documentary series Islam Unveiled (2004) exploring Muslim feminism. Samira has also worked as a news anchor for Deutsche Welle TV in Berlin and writes regularly on culture and politics for newspapers and magazines including The Guardian and The Independent.&lt;/p&gt;</prose>
										<image/>
										<image_caption/>
										<quote/>
										<quote_attribution/>
										<clip/>
										<media_position>left</media_position>
									</content>
								</form>
							</document>
						</result>
					</content_block>
				</content_blocks>
				<onward_journeys/>
			</profile>
		</form>
	</document>
</result>
```
</details>

<details>
  <summary>XML for parent IN PROGRESS currently but published in the past</summary>

```xml



<!-- 
THIS XML IS AN EXAMPLE OF AN INDIVIDUAL PROFILE PUBLISHED BUT THIS TIME
THE PARENT PROFILE WAS PUBLISHED IN THE PAST EVEN IF RIGHT NOW IS IN PROGRESS
-->



<?xml version="1.0" encoding="UTF-8"?>
<result xmlns="https://production.bbc.co.uk/isite2/contentreader/xml/result">
	<metadata>
		<modifiedDateTime>2018-08-15T15:30:12.014912Z</modifiedDateTime>
		<creationDateTime>2014-05-20T13:50:45.687017Z</creationDateTime>
		<type>programmes-profile</type>
		<fileId>programmes-profile-b006tp43-samiraahmed</fileId>
		<guid>5531af13-a351-3960-93d1-193ec85bfb95</guid>
	</metadata>
	<document>
		<form xmlns="https://production.bbc.co.uk/isite2/project/progs-radio3/programmes-profile">
			<metadata>
				<name>Samira Ahmed (b006tp43)</name>
				<title>Samira Ahmed updated (ccccc... in prgress)</title>
				<type>individual</type>
				<short_synopsis>asdfasdfasdfasdf</short_synopsis>
				<long_synopsis>Samira Ahmed is an award winning journalist with 20 years experience in print and broadcast and a Visiting Professor of Journalism at Kingston University.</long_synopsis>
				<image>p00qsl1w</image>
				<position>0</position>
				<tagline/>
				<group_size/>
				<parent_pid>b006tp43</parent_pid>
				<branding_id/>
				<file_id>b006tp43-samiraahmed</file_id>
				<contributor_pid/>
				<contribution_pid/>
				<parents>
					<parent>
						<result xmlns="https://production.bbc.co.uk/isite2/contentreader/xml/result">
							<metadata>
								<modifiedDateTime>2018-08-15T14:16:05.865103Z</modifiedDateTime>
								<creationDateTime>2018-08-15T12:44:43.343444Z</creationDateTime>
								<type>programmes-profile</type>
								<fileId>programmes-profile-1534340118</fileId>
								<guid>7d554868-17fd-4019-bd99-149b39e87a11</guid>
							</metadata>
							<document> <!-- PARENT IN PROGRESS BUT IT WAS ONCE PUBLISHED IN THE PAST -->
								<form xmlns="https://production.bbc.co.uk/isite2/project/progs-radio3/programmes-profile">
									<metadata>
										<name>ONE: in progress individual profile </name>
										<title>SECOND a parent profile that is supposed not be published ever</title>
										<type>individual</type>
										<short_synopsis>Some short </short_synopsis>
										<long_synopsis>asdfaiosdfpoaisdjfpoasidjfpo ijasdpfoi jasdpofij aspdfj asdfj asdf jasd ofjasdpofjapoisdfjapoisdjfapoidsjfpoasdijfasdf</long_synopsis>
										<file_id>1534340118</file_id>
										<image>p00qsl1w</image>
										<tagline/>
										<position>0</position>
										<group_size/>
										<parent_pid>b006tp43</parent_pid>
										<branding_id/>
										<contributor_pid/>
										<contribution_pid/>
										<bbc_site/>
									</metadata>
									<key_facts/>
									<profile>
										<content_blocks>
											<content_block/>
										</content_blocks>
										<image_portrait/>
										<onward_journeys/>
									</profile>
								</form>
							</document>
						</result>
					</parent>
				</parents>
				<bbc_site/>
			</metadata>
			<key_facts/>
			<profile>
				<image_portrait/>
				<content_blocks>
					<content_block>
						<result xmlns="https://production.bbc.co.uk/isite2/contentreader/xml/result">
							<metadata>
								<modifiedDateTime>2014-06-06T13:58:47.22545Z</modifiedDateTime>
								<creationDateTime>2014-05-20T13:50:45.687017Z</creationDateTime>
								<type>programmes-content-prose</type>
								<fileId>programmes-content-b006tp43-samiraahmed-profile-prose-1</fileId>
								<guid>73a2bd99-0be1-3df0-8cf8-050aa660c0d8</guid>
							</metadata>
							<document>
								<form xmlns="https://production.bbc.co.uk/isite2/project/progs-radio3/programmes-content-prose">
									<metadata>
										<name>Samira Ahmed (b006tp43) - Prose 1</name>
										<file_id>b006tp43-samiraahmed-profile-prose-1</file_id>
									</metadata>
									<content>
										<title/>
										<prose>&lt;p&gt;She has fronted The Proms on BBC4, presents Sunday Morning Live and Newswatch on BBC1 and writes a monthly column in The Big Issue. Samira began her career as a BBC news trainee in 1990. She covered the OJ Simpson case as BBC Los Angeles Correspondent, where she made a documentary about the history of the city's cinemas. Her films as a presenter and reporter at Channel 4 News from 2000 to 2011 included assessing social mobility on screen since the 1960s with Julie Walters and Douglas Trumbull's views on ecological disaster from Silent Running to the BP Gulf of Mexico disaster.&lt;/p&gt;&lt;p&gt;She wrote and presented the acclaimed Channel 4 documentary series Islam Unveiled (2004) exploring Muslim feminism. Samira has also worked as a news anchor for Deutsche Welle TV in Berlin and writes regularly on culture and politics for newspapers and magazines including The Guardian and The Independent.&lt;/p&gt;</prose>
										<image/>
										<image_caption/>
										<quote/>
										<quote_attribution/>
										<clip/>
										<media_position>left</media_position>
									</content>
								</form>
							</document>
						</result>
					</content_block>
				</content_blocks>
				<onward_journeys/>
			</profile>
		</form>
	</document>
</result>
```
</details>

## Example URL:
http://sandbox.bbc.co.uk/programmes/profiles/20cgQm4fNTGVprqDRPDfD8J/samira-ahmed-updated-ccccc-in-prgress?__flush_cache=1

You can find the child profile in Isite (which includes also the parent) in here:
https://production.test.bbc.co.uk/isite2-xforms/fr/progs-radio3/programmes-profile/edit/5531af13-a351-3960-93d1-193ec85bfb95

### CODE REVIEW

**Comment by programmes-developers** [comment]
Can one of the admins verify this patch?

**Comment by votemike** [comment]
Do you have example sandbox URLs where we can verify the fix?

File: src/ExternalApi/Isite/Mapper/ProfileMapper.php [hunk 1]
```diff
@@ -38,7 +38,9 @@ public function getDomainModel(SimpleXMLElement $isiteObject): Profile
         $parents = [];
         if (!empty($formMetaData->parents->parent->result)) {
             foreach ($formMetaData->parents as $parent) {
-                $parents[] = $this->mapperFactory->createProfileMapper()->getDomainModel($parent->parent->result);
+                if (empty($parent->parent->result->document)) {
```
Comment (votemike) [inline src/ExternalApi/Isite/Mapper/ProfileMapper.php]: If it's empty, add it?

Comment (votemike) [inline src/ExternalApi/Isite/Mapper/ProfileMapper.php]: `if ($this->isPublished($parent->parent)) {` ?

**Pull request merged**